File: cpp/621_任务调度器_medium.cpp
class Solution {
public:
    int leastInterval(vector<char>& tasks, int n) {
        int maxCount = 0;
        int N = tasks.size();
        
        unordered_map<char, int> tasks_map;
        for(char task : tasks){
            tasks_map[task]++;
            maxCount = max(maxCount, tasks_map[task]);
        }
        int maxNum = 0;
        for(unordered_map<char, int> ::iterator iter=tasks_map.begin(); iter!=tasks_map.end(); iter++)
            if(iter->second == maxCount)
                maxNum++;
        return max( (maxCount-1) * (n+1) + maxNum, N );

    }
};

File: python/621_任务调度器_medium.py
class Solution:
    def leastInterval(self, tasks: List[str], n: int) -> int:
        tasks_map = collections.Counter(tasks)
        maxCount = max(tasks_map.values())
        maxSum = sum([1 for i in tasks_map.values() if i==maxCount])
        return max((maxCount-1) * (n+1) + maxSum, len(tasks))